relay-server/src/services/project_cache.rs

```diff
@@ -859,17 +859,19 @@ impl ProjectCacheBroker {
     }
 
     fn handle_merge_buckets(&mut self, message: MergeBuckets) {
+        let project_cache = self.services.project_cache.clone();
         let aggregator = self.services.aggregator.clone();
         let outcome_aggregator = self.services.outcome_aggregator.clone();
         let envelope_processor = self.services.envelope_processor.clone();
-        // Only keep if we have an aggregator, otherwise drop because we know that we were disabled.
-        self.get_or_create_project(message.project_key())
-            .merge_buckets(
-                aggregator,
-                outcome_aggregator,
-                envelope_processor,
-                message.buckets(),
-            );
+
+        let project = self.get_or_create_project(message.project_key());
```

Comment:
I wonder if we should just always prefetch in `get_or_create_project`

Reply:
@Dav1dde I would even go further and unify the multiple state enums we have for project configs at the moment:

* `GetOrFetch` - either a cached project config or a channel
* `State` - either a cached project config or a metrics buffer
* `ExpiryState` - either `Updated`, `Stale`, or `Expired`

We might be able to make a single enum out of these, and give them an interface that makes it impossible to access a project without prefetching.